exonum/src/runtime/mod.rs

```diff
@@ -255,9 +258,23 @@ pub trait Runtime: Send + Debug + 'static {
         tx_sender: &ApiSender,
     );
 
-    fn services_api(&self) -> Vec<(String, ServiceApiBuilder)> {
+    /// Collect the full list of API handlers from the runtime for the built-in Exonum API server.
+    ///
+    /// This method is called during the API server restart. Use this method if you do not plan to
+    /// use your own API processing mechanism.
+    ///
+    /// Warning! It is a temporary method which retains the existing `RustRuntime` code.
+    /// It will be removed in the future.
+    #[doc(hidden)]
+    fn api_endpoints(&self, _context: &ApiContext) -> Vec<(String, ServiceApiBuilder)> {
         Vec::new()
     }
+
+    /// Notify the runtime about the changes in the list of service instances.
+    ///
+    /// The purpose of this method is to provide building blocks to create your own
+    /// API processing mechanisms.
+    fn notify_api_changes(&self, _context: &ApiContext, _changes: &[ApiChange]) {}
```

Comment:
What shall the runtime implementation do if some services fail to mount the API (e.g., an Exception occurs)? How does it communicate that to the core? Shall this error affect other services-to-be-connected?

Shan't we instead have a method notifying that _a single_ service is to be connected or disconnected from the web server? Otherwise the communication of potential errors in API configuration becomes complicated.